ERCS/erc-7769.md

```diff
@@ -0,0 +1,543 @@
+
+The result SHOULD be set to the `userOpHash` if and only if the request passed simulation and was accepted
+in the client's UserOperation pool.
+
+If the validation, simulation, or UserOperation pool inclusion fails,
+`userOpHash` SHOULD NOT be returned. Rather, the client SHOULD return the failure reason.
+
+##### Parameters:
+
+1. **UserOperation** a full user-operation struct.\
+  All fields MUST be set as hex values.\
+  Empty `bytes` block (e.g. empty `initCode`) MUST be set to `"0x"`\
+2. **factory** and **factoryData**\
+  Must provide either both of these parameters, or none.
+3. **paymaster**, **paymasterData**, **paymasterValidationGasLimit**, **paymasterPostOpGasLimit**\
+  Must provide either all of these parameters, or none.
+4. **entryPoint** the `EntryPoint` contract address the request should be sent through.\
+  This MUST be one of the entry points returned by the `supportedEntryPoints` RPC call.
+
+##### Return value:
+
+* If the UserOperation is valid, the client MUST return the calculated `userOpHash` for it
+* in case of failure, MUST return an `error` result object, with `code` and `message`.\
+  The error code and message SHOULD be set as follows:
+  * **code: -32602** - invalid `UserOperation` struct/fields
+  * **code: -32500** - transaction rejected by `EntryPoint` contract's `simulateValidation` function
+    during wallet creation or validation
+    * The `message` field MUST be set to the emitted `FailedOp` event's "`AAxx`" error message from the `EntryPoint`
+  * **code: -32501** - transaction rejected by `paymaster` contract's `validatePaymasterUserOp` function
+    * The `message` field SHOULD be set to the revert message from the `paymaster` contract
+    * The `data` field MUST contain a `paymaster` value
+  * **code: -32502** - transaction rejected because of [ERC-7562](./eip-7562) opcode validation rule violation
+  * **code: -32503** - UserOperation out of time-range:\
+    either wallet or paymaster returned a time-range, and it has already expired or will expire soon.
+    * The `data` field SHOULD contain the `validUntil` and `validAfter` values
+    * The `data` field SHOULD contain a `paymaster` address if this error was triggered by the `paymaster` contract
+  * **code: -32504** - transaction rejected because `paymaster` is throttled or banned due to ERC-7562 reputation rules
+    * The `data` field SHOULD contain a `paymaster` address
+  * **code: -32505** - transaction rejected because `paymaster` contract's ERC-7562 stake or unstake-delay is too low
+    * The `data` field SHOULD contain a `paymaster` address
+    * The `data` field SHOULD contain a `minimumStake` and `minimumUnstakeDelay`
+  * **code: -32507** - transaction rejected because of wallet signature check failed
+  * **code: -32508** - transaction rejected because paymaster balance can't cover all pending `UserOperations`.
+
+##### Example:
+
+Request:
+
+```json
```

Comment:
````suggestion
```js
````

Not a hard requirement, but if you use `js` instead of `json`, it'll still highlight but won't complain about comments.